### ISSUE TITLE
force📦beta:version:create incorrectly handles profiles in unpackaged metadata

### ISSUE THREAD

**Issue by telennar-github** [forcedotcom/cli#1846]
### Summary

We have specified unpackaged metadata in our sfdx-project file This package includes a Profile needed for our unit tests. A few weeks ago, I successfully built and promoted the project as a managed package. Yesterday, I went to build the next version of our package and received an error about not being able to remove components from 2GP packages.

Moving the profile under the force-app directory allowed the package to build successfully, but we don't want that Profile to be installed into customer orgs.

This may have something to do with the previous version being promoted to released. We created almost a dozen beta packages without error. This is the first we've tried to create after a version was promoted.

### Steps To Reproduce:

Our code cannot be released due to intellectual property restrictions.

1. Create a project with an unpackaged metadata directory.
2. Place a Profile under unpackaged metadata.
3. Build a 1.0 beta package.
4. Promote the package to released.
5. Build the a 2.0 beta package.

### Expected result

The package should have been created successfully.

### Actual result

The package creation failed with an error about metadata removed from the package.

### System Information

- Which shell/terminal are you using? Both Powershell and cmd
```
sfdx version --verbose --json
{
  "cliVersion": "sfdx-cli/7.180.0",
  "architecture": "win32-x64",
  "nodeVersion": "node-v18.12.1",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 1.3.6 (core)",
    "@oclif/plugin-commands 2.2.1 (core)",
    "@oclif/plugin-help 5.1.19 (core)",
    "@oclif/plugin-not-found 2.3.9 (core)",
    "@oclif/plugin-plugins 2.1.7 (core)",
    "@oclif/plugin-search 0.0.4 (core)",
    "@oclif/plugin-update 3.0.7 (core)",
    "@oclif/plugin-version 1.1.3 (core)",
    "@oclif/plugin-warn-if-update-available 2.0.14 (core)",
    "@oclif/plugin-which 2.2.2 (core)",
    "alias 2.1.10 (core)",
    "apex 1.3.0 (core)",
    "auth 2.3.0 (core)",
    "community 2.0.18 (core)",
    "config 1.4.22 (core)",
    "custom-metadata 2.0.5 (core)",
    "data 2.1.13 (core)",
    "generator 2.0.12 (core)",
    "info 2.2.10 (core)",
    "limits 2.1.2 (core)",
    "org 2.2.15 (core)",
    "packaging 1.10.9 (core)",
    "schema 2.1.17 (core)",
    "signups 1.2.12 (core)",
    "source 2.3.7 (core)",
    "telemetry 2.0.5 (core)",
    "templates 55.1.0 (core)",
    "trust 2.1.0 (core)",
    "user 2.1.21 (core)",
    "@salesforce/sfdx-plugin-lwc-test 1.0.1 (core)",
    "salesforce-alm 54.8.4 (core)"
  ],
  "osVersion": "Windows_NT 10.0.22621",
  "shell": "cmd.exe",
  "rootPath": "C:\\Users\\User\\AppData\\Local\\sfdx\\client\\7.180.0-d902059"
}
```
### Additional information
```
sfdx force:package:beta:version:create --codecoverage --installationkeybypass --wait=60 --package=0Ho3s000000GmeGCAS
Version create.... Create version status: Error
ERROR running force:package:beta:version:create:  Multiple errors occurred:
(1) Can’t create package version. Removing metadata components from a second-generation managed package is a pilot feature that requires acceptance into a pilot program.

To join the Remove Metadata Components from 2GP pilot, file a case in the Salesforce Partner Community.

To create a package version without removing metadata components, restore the removed components and try creating the package version again. Add components {Profile=[No Access Profile]} back into the package version, and create the package version again.
```

**Comment by iowillhoit** [comment]
Hey @telennar-github, thanks for the Issue. At a glance this looks like it might need to be redirected to the Packaging team. 
A few questions:
- Are you able to complete this successfully with the non-beta commands? 
- How about with the `beta` commands on a slightly older version of the cli, say `7.177.1`?

**Comment by telennar-github** [comment]
Hi Wilhoit.

- I've tried both the beta and non-beta commands and both give the same result. As an aside, the original release package was built with the non-beta commands. I tried both on the new one when I had issues; neither worked.
- I think I originally tried on 7.179(?), but I don't recall for certain. I updated to the latest after the problem occurred in the hopes that would solve the issue. If you think it might make a difference, is there a way to downgrade to 1.177.1?

**Comment by iowillhoit** [comment]
Ok, if this is not working on the old commands then this would be a Packaging team question.

**Comment by iowillhoit** [comment]
Gotcha, if it is not working with the old commands, then this will need to be redirected to the Packaging team. I'll share a few other things to try and I'll also add a Github label that will provide instructions to open a support case if those do not help.

A few notes: 
- It might be worth updating to the newest version released today (`7.181.1`), It has some updates to `force:package:beta:version:create:report` that improves error handling. Details in the [release notes](https://github.com/forcedotcom/cli/tree/main/releasenotes/sfdx#71811-dec-15-2022-stable).
- In the `Additional information` you provided above, it looks like this is a pilot-feature and you could gain access to use it by requesting to join the "Remove Metadata Components from 2GP" pilot.
- Regarding downgrading, it depends on how you installed the CLI 
    - If you installed with `npm`, it is as simple as running `npm install sfdx-cli@THE_VERSION -g` 
    - Links for older OS Installers and Tarballs [can be found here](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli.htm#sfdx_setup_install_cli_olderversions)

Best of luck getting this sorted out!

**Comment by daveespo** [comment]
There was a post by Dileep about how Profiles are aggregated during the package build process. See his "Feb 28, 2019, 4:14 PM" comment on this thread:

https://trailhead.salesforce.com/trailblazer-community/feed/0D53A000040SePr

We found out about this behavior when we tried to move a Profile out of the package directory into the unpackagedMetadata directory .. and we got the same error you got about not being able to delete components from 2GP without being in the pilot. 

It isn't visible to us that a prior package version build had created a hard managability rule around the Profile even though that Profile wasn't part of the package metadata when installed into the client org.

Dileep's comment was created prior to addition of the unpackagedMetadata feature .. but I suspect that's what's going on even in your case.

**Comment by tyschacht** [comment]
@telennar-github  -- Did you ever figure out a solution to this?   We're running into the same issue and it seems the only reasonably quick way around it might be to opt-in to the "Metadata Delete" pilot.

**Comment by telennar-github** [comment]
Hi @tyschacht,

Take a look at [Remove Metadata Components from Second-Generation Managed Packages (Generally Available)](https://help.salesforce.com/s/articleView?id=release-notes.rn_sfdx_packaging_remove_components.htm&type=5&release=236) and [Remove Components from First-Generation Managed Packages
](https://developer.salesforce.com/docs/atlas.en-us.packagingGuide.meta/packagingGuide/packaging_managed_component_deletion.htm) for detailed information. We had them grant us permission to delete manage package components in our DevHub. We had several issues running in parallel, so I could be wrong, but I think that  was what fixed the issue.

Regards.

**Comment by shetzel** [comment]
This issue is being tracked internally by the packaging team with W-9186045.  It mentions that a workaround is to build the version with the `--codecoverage` flag.

**Comment by jvg123** [comment]
@shetzel can you please link to that internal issue you referenced, or copy the entire workaround?

**Comment by iowillhoit** [comment]
@jvg123 I scanned that internal ticket and I did not see a mention of the `--codecoverage` workaround. However, that ticket is closed and [a PR was merged](https://github.com/forcedotcom/packaging/pull/265) a month ago. It looks like the fix is included starting in `plugin-packaging` version `1.16.10`. 

Could you make sure that you're use that version (or higher) and let me know if it's working now? Thanks

**Comment by jvg123** [comment]
Hi @iowillhoit thanks for your help!

Unfortunately we're still unable to create a package version even with `packaging` version `1.16.10`

But looking at the PR it sounds like it does not really address our issue, which is exactly the one described in this ticket.

The PR will stop the CLI from packaging profiles from unpackaged metadata dirs. But in our case it seems the damage is already done: the unwanted profile was already packaged in a previous version due to the existence of this defect in an older CLI version.

It would be great if the new CLI could auto-magically detect this situation and reverse it -- but I understand that would be tricky.

So I think the only way forward for us is to enable the "Remove Metadata Components from Second-Generation Managed Packages" feature so we can remove the phantom profile.

**Comment by cristiand391** [comment]
Closing this since the CLI bug was fixed.
As jvg123 said, you can request to get access to the feature and remove the metadata from your 2GP (see the list of supported metadata):

https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_dev2gp_remove_md_components.htm
